### ISSUE TITLE
Google let's anyone use Google Earth buildings as 3D Tiles

### ISSUE THREAD

**Issue by donnyv** [maplibre/maplibre-gl-js#2509]
With Google changing their policy and letting anyone include their Google Earth buildings as GlTF. Maybe Maplibre can add it as a native layer source.  
https://cloud.google.com/blog/products/maps-platform/create-immersive-3d-map-experiences-photorealistic-3d-tiles

**Comment by HarelM** [comment]
Can you clarify the issue? Problem statement? Feature request?
Do you mean that maplibre should support 3D tiles as part of this issue?

**Comment by IvanSanchez** [comment]
Your premise ("Google let's[sic] anyone use Google Earth buildings as 3D Tiles") is wrong. Let me quote the linked press release, emphasis mine:

> Today we're launching the Experimental release of Photorealistic 3D Tiles, **available through the Map Tiles API**.

Now let me quote their [3D tiles documentation](https://developers.google.com/maps/documentation/tile/3d-tiles):

> Before you start using the Photorealistic 3D Tiles API, you need a project with a billing account, and you need the Map Tiles API enabled.

That is not "letting anyone", and there has been no policy change. Please stop the spread of openwashing.

**Comment by donnyv** [comment]
@IvanSanchez Sorry I didn't mean too to spread "openwashing". Which until now I didn't know that word existed LOL I didn't realize the data wasn't actually open sourced. Just free to use there service....for now.

As for support for 3D Tiles. I still think it would be beneficial to have it in maplibra.  
https://github.com/CesiumGS/3d-tiles

**Comment by prozessor13** [comment]
Beside google, other cities, like vienna, offers 3dtiles (https://www.wien.gv.at/stadtplan3d/ click on Content and then activate 3D-Mesh). So I think this would be really a great addition to maplibre, too.
But implementing 3dtiles/gltf into maplibre will be, for sure, a big project. May it is better to render it with deck.gl, beside with maplibre. Deck.gl for 3dTiles, maplibre for labels and markers.

Would be great to start a discussion, especially some thoughts by people who know deck.gl, how a potential implementation could look like.

**Comment by wipfli** [comment]
I wonder what the best approach is for 3D content in MapLibre. Already now, you can integrate single glTF models in MapLibre GL JS with Three.JS or deck.gl.

But that is just single 3D objects with a fixed level of detail, a bit like Apple maps does it in the vector version or also Snap if I am not wrong.

**Comment by prozessor13** [comment]
I agree that 3DTiles are a bit of a monster. I had a requirement to render CityGML Data onto maplibre, and i ended up reusing the VectorTile spezification with a custom geometry-type with 3 coordinates instead of 2d. This had the advantage the all the tile-covering logic of maplibre can be reused. But in the end it is a custom solution which is not future-proof. So, beacuse now "everyone" is using 3DTiles/gltf I think there is no way around. I also do not know some kind of lightweight library to load/parse 3DTiles, and because deck.gl has already an implementation, why not use it?

If you want to have a look onto my custom-3d-vectortiles CityGML Demopage:
   https://static.maptoolkit.net/maplibre/buildings.html#15.23/48.210618/16.372333/-20/74
Except the lack of render textures onto the 3d objects, it is really performant, and was not very much work implementing/integrating it into maplibre.

**Comment by wipfli** [comment]
Looks amazing, thanks for sharing, @prozessor13 !

**Comment by StylianosVitalis-TomTom** [comment]
@prozessor13 This is awesome! It performs impressively, as well!

If I get it correctly, then you had to implement your own exporter and MapLibe GL JS parser for this customized 3D vector tiles?

**Comment by prozessor13** [comment]
Sorry, this is a product of our company :) I only wanted to demonstrate a way beside 3DTiles, and for sure this format could be sufficient to render 3d-meshes (may even more performant and memory-friendly), and i really would like to implement this, but because everyone else uses 3DTiles, i think this is not the future.

**Comment by StylianosVitalis-TomTom** [comment]
Absolutely! Still, impressive work. Top notch! 🙌

I remain of the opinion that the whole 3D Tiles specification is an overkill (for non-mesh data, at least). It fits the Google use case, but for data such as in your example I think there are elements of 3D Tiles we can reuse (like the glTF+extension to hold attributes), but the tile definition specification is adding too much complexity, both to the implementation and the data processing part.

**Comment by wipfli** [comment]
If someone wants to implement 3d tiles in MapLibre, then in principle there is some funding available from cesium via the ecosystem grants program: https://cesium.com/cesium-ecosystem-grants/

**Comment by jsehner** [comment]
LoD2 buildings as 3D Tiles directly in MapLibre would be great! This addon works well, which was developed for Mapbox GL JS, but also works fine with MapLibre: https://github.com/Geodan/mapbox-3dtiles. It was use for the visualisation of Germany for basemap.de. A beta-service is released for Cesium (buildings on terrain) and MapLibre (buildings on sea level). Simple Viewer: https://web3d.basemap.de/maplibre/lod2.html#16.3/50.941485/6.959092/-39.2/57. Description: https://basemap.de/beta-lod2/. What is missing: It doesn't work in combination with the 3D Terrain, only with flat terrain. By the way: A different colour for the roofs is important in my opinion.

**Comment by gberaudo** [comment]
A few remarks:
- 3d-tiles is an OGC standard;
- [3d-tiles next](https://github.com/CesiumGS/3d-tiles/tree/main/next) is the next iteraction of the standard, which will include some simplifications (direct use of gltf, implicit tiling scheme, ...).
